### ISSUE TITLE
CIFAR10 Tutorial "Model Diverged with loss = NaN"

### ISSUE THREAD

**Issue by NiftyGrimoire** [ROCmSoftwarePlatform/hiptensorflow#39]
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
None
### Environment info
Operating System: Ubuntu 16.04
Followed Installation steps in: https://github.com/ROCmSoftwarePlatform/hiptensorflow/blob/hip/rocm_docs/tensorflow-install-basic.md 
Also tried docker: https://hub.docker.com/r/rocm/hiptensorflow/
To same results.

### 2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.
1.0.1

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

The CIFAR-10 dataset tutorial.

### What other attempted solutions have you tried?

I have tried playing with the learning rate of the demo, but I hesitate to make any more major changes. I assume the tutorials should work out of the box?

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

**Comment by NiftyGrimoire** [comment]
Just tested with a learning rate of zero, still diverges.
I have also attempted changing from distorted_inputs() to inputs() to no avail.
Running the ResNet model appears to work (Though ResNet is slower, so I have yet to achieve the same number of steps).

The diverging occurs randomly, at points in the code. I returns the NaN error anywhere between 400 to 90000 steps in. (Furthest I have gotten without diverging).

**Comment by bensander** [comment]
A bit of a guess but can you try changing the batch_size?  We have another ticket related to NANs in cifar10 training which occurs only for specific batch sizes.

**Comment by NiftyGrimoire** [comment]
Will try that now. Will update soon.

**Comment by NiftyGrimoire** [comment]
@bensander 
Just tested once with batch size of 64. Still diverges (again, zero learning rate). Running batch size of 100 now. 

In the middle of modifying the cifar10 CNN to accept SVHN data (as per the tutorials outlined on the Tensorflow site). I'll report on that as well. 

Just a quick shout, thanks for all your, and everyone else's, hard work on porting these platforms to hip/ROCm. Makes dipping my feet into this field much easier.

**Comment by bensander** [comment]
@7SK - Sandeep can you repro this one internally with pre-ROCm 1.7 bits?
@NiftyGrimoire - we have made several improvements to TF and the stack since the 1.6.4 branch.   Many o the user bits for 1.7 are already publicly available in source form.  If you are brave, you can rebuild HCC, HIP, and the hip-develop branch of TensorFlow.  Or we should have this avail in prebuilt packages in ~week.

**Comment by NiftyGrimoire** [comment]
I can give that a bash, though that will have to wait until tomorrow I'm afraid. 
As an update, it does seem to still diverge at 100. 

Since it seems some other people was able to get through the tutorials no problem, is there any hardware information I should provide? I'm using a RX 480 8Gb. Driver is amdgpu-pro-17.40.

Implemented version with SVHN data. Not sure if my implementation is correct, so take this with a grain of salt, but I'm able get the training started, but that also diverges. 

As an aside, I infrequently get "Page not present or supervisor privilege." (see https://github.com/RadeonOpenCompute/ROCR-Runtime/issues/20) But I believe thats an unrelated issue.

**Comment by parallelo** [comment]
Hi @NiftyGrimoire, 

Specifically for machine intelligence workloads, we recommend switching away from the amdgpu-pro drivers.  At this point, you'll need the ROCm stack for TF support as shown in the documentation.  We'll also update the TF documentation to explicitly state this requirement.  

Please uninstall amdgpu-pro and follow the ROCm installation instructions here:  https://rocm.github.io/install.html

Feel free to keep us posted on how it goes.  

Best, 

Jeff

**Comment by NiftyGrimoire** [comment]
Oops, apologies for that. Thank you for the advice. I'll will try that now. Been unable to test the devel branch I'm afraid. Time to test with TF has been spotty recently.

**Comment by parallelo** [comment]
Hi @NiftyGrimoire - Any luck?  (Your statement regarding containers was correct: you'll need to install the ROCm drivers on the host, and then the user-level components and the thunk inside the container)

**Comment by whchung** [comment]
Please kindly try https://github.com/ROCmSoftwarePlatform/tensorflow/ and open the issue there in case it persists. This project has been deprecated and I'll close this repository very soon. Thanks for your understanding.
